src/vendor/models/profile.py

```diff
@@ -210,24 +210,36 @@ def get_active_subscriptions(self):
         return self.subscriptions.filter(status=SubscriptionStatus.ACTIVE)
 
     def get_next_billing_date(self):
+        '''Returns the next billing date for the customers subscriptions'''
         next_billing_dates = []
         
         if not self.subscriptions.filter(status=SubscriptionStatus.ACTIVE).count():
             return None
 
         next_billing_dates = [subscription.get_next_billing_date() for subscription in self.get_active_subscriptions()]
+        
+        if not next_billing_dates:
+            return None
+        
+        filtered_dates = [date for date in next_billing_dates if date is not None]
```

Comment:
should the return case come after calculating filtered_dates? what if `next_billing_dates == [None]`?
```
if not filtered_dates:
    return None
```